be explicit about supported RN versions

vonovak opened the pull request:
motivation: readme contains information about supported versions and it should be more explicit because: 

* Currently is says "React Native versions 0.44 and above are supported." and the readme wasn't updated in 5 months, so it certainly wasn't correct in the time when 0.50 wasn't working on android. 
* It also isn't "future-proof" because whenever a new RN version is released, the sentence automatically makes user think the new version is supported which may not be true

For people like me who occasionally come to the repo to check if their RN version is supported, this will be a big help that will save time digging through issues.

I based the table contents on this [comment](https://github.com/wix/detox/issues/608#issuecomment-388679980), not sure what the supported versions really are.

noomorph commented:
Might be related to https://github.com/wix/detox/issues/774

rotemmiz commented:
Instead of using RN as a point of reference, let's use platform as point of reference.
Documentation is still missing data on what APIs exist on iOS and what APIs exist on Android. I intend to create such a list in the near future, and I want yours to align with it.

| iOS | Android
|-----|-------------
| >=0.55 | <=0.51

vonovak commented:
thanks for the comment @rotemmiz, you table structure makes much more sense. I have changed the PR in accordance to it., I just switched the inequality sign in the iOS column because I believe it should be <=0.55 (?).

If there are some more tricks in the supported versions, we can further break the table by also including detox version column on the left.

rborn commented:
@markhu @rotemmiz @vonovak  I have `"detox": "^7.4.0"` running with `"react-native": "0.55.4"` on android

I will test with `"detox": "^8.0.0"` and update you.

rborn commented:
@markhu @rotemmiz @vonovak 
I updated to `8.0.0` but I get an error:
~~~
 11: running "/Work/sdks/android-sdk-macosx/platform-tools/adb -s 192.168.57.101:5555 shell ps -AMo NAME,PID" returned undefined
~~~

It doesn't seem to be related to the ReactNative version though.

rotemmiz commented:
This is unrelated to this PR.
Please open a new issue with full logs for that.
Moreover, add your ADB version, emulator version and device API level.
